crypto/cmp/cmp_vfy.c

```diff
@@ -0,0 +1,868 @@
+            break;
+        }
+        /* validate sender name of received msg */
+        if (msg->header->sender->type != GEN_DIRNAME) {
+            CMPerr(0, CMP_R_SENDER_GENERALNAME_TYPE_NOT_SUPPORTED);
+            break; /* FR#42: support for more than X509_NAME */
+        }
+        /*
+         * Compare actual sender name of response with expected sender name.
+         * Mitigates risk to accept misused certificate of an unauthorized
+         * entity of a trusted hierarchy.
+         */
+        if (ctx->expected_sender != NULL) {
+            /* set explicitly or subject of ctx->srvCert */
+            X509_NAME *sender_name = msg->header->sender->d.directoryName;
+            if (X509_NAME_cmp(ctx->expected_sender, sender_name) != 0) {
+                CMPerr(0, CMP_R_UNEXPECTED_SENDER);
+                add_name_mismatch_data(NULL, ctx->expected_sender, sender_name);
+                CMPerr(0, CMP_R_UNEXPECTED_SENDER);
+                add_name_mismatch_data(NULL, sender_name, ctx->expected_sender);
+                break;
+            }
+        }/* Note: if recipient was NULL-DN it could be learned here if needed */
+
+        scrt = ctx->srvCert;
+        if (scrt == NULL)
+            scrt = find_srvcert(ctx, msg);
+        if (scrt != NULL) {
+            if (verify_signature(ctx, msg, scrt))
+                return 1;
+
+            /* failure; add further cert matching & validation diagnostics: */
+            if (ctx->srvCert != NULL)
+                (void)cert_acceptable(scrt, msg, ctx->trusted);
+
+            /*
+             * potentially the server cert finding algorithm took wrong cert:
+             * the server certificate may not have a subject key identifier
+             * (including such in EE certs is only a "SHOULD" requirement in
+             * RFC 5280, section 4.2.1.2) or the CMP server is not conforming
+             * with RFC 4210 and is failing to set senderKID although having
+             * several keys associated with its name
+             */
+            if (!X509_find_by_issuer_and_serial(msg->extraCerts,
+                                                X509_get_issuer_name(scrt),
+                                                (ASN1_INTEGER *)
+                                                X509_get0_serialNumber(scrt)))
+                ossl_cmp_add_error_line("  certificate used for signature verification attempt was not found in extraCerts");
+
+            if (msg->header->senderKID == NULL) {
+                ossl_cmp_add_error_line("  no senderKID in CMP header; risk that correct server cert could not be identified");
+            } else { /* server cert should match senderKID in header */
+                if (!check_kid(scrt, msg->header->senderKID))
+                    /* here this can only happen if ctx->srvCert has been set */
+                    ossl_cmp_add_error_line("  for senderKID in CMP header there is no matching Subject Key Identifier in context-provided server cert");
```

Comment:
Excessively long lines here and on lines above

Reply:
from https://www.openssl.org/policies/codingstyle.html , Chapter 2: 

... Statements longer than 80 columns must be broken into sensible chunks, unless exceeding 80 columns significantly increases readability and does not hide information. ... Never break user-visible strings, however, because that breaks the ability to grep for them.

Is this still valid?

Reply:
Hmmmm....it does still say that, although I'm not entirely sure I agree with it. But ok, leave it for now.